src/gs-screenshot-image.c

```diff
@@ -250,18 +337,30 @@ gs_screenshot_image_complete_cb (SoupSession *session,
 			return;
 		}
 	} else {
+		AsImageKind kind;
+
 		/* save to file, using the same code as the AppStream builder
 		 * so the preview looks the same */
 		im = as_image_new ();
 		as_image_set_pixbuf (im, pixbuf);
+
 		ret = as_image_save_filename (im, ssimg->filename,
 					      ssimg->width * ssimg->scale,
 					      ssimg->height * ssimg->scale,
-					      AS_IMAGE_SAVE_FLAG_PAD_16_9, &error);
+					      AS_IMAGE_SAVE_FLAG_PAD_16_9,
+					      &error);
 		if (!ret) {
 			gs_screenshot_image_set_error (ssimg, error->message);
 			return;
 		}
+
+		/* if the kind is unknown then we save also the counterpart size
+		 * image in order to save an extra download when the user
+		 * requests that size later */
```

Comment:
Why does the "unknown" kind imply that we want to save both versions? Also I'm a bit confused why we need to do this at all. Won't it potentially upscale images, making them look bad?